web/src/lib/components/admin-page/settings/setting-select.svelte

```diff
@@ -27,13 +28,13 @@
 	</div>
 	<select
 		class="immich-form-input w-full"
-		name="presets"
-		id="preset-select"
+		{name}
+		id="{name}-select"
 		bind:value
 		on:change={handleChange}
 	>
 		{#each options as option}
-			<option value={option}>{option}</option>
+			<option value={option.value}>{option.text}</option>
```

Comment:
Why is it necessary to change this to have a separate value and text, I believe those two values are the same in your usage?

Reply:
This separation allows to have a short `value` (used as a key to identify selected option in code and stored in the database) and long `text` for the user (that maybe eventually should be localized). For example, in my case one of transcoding presets has `value="optimal"` and `text="Videos higher than 1080p or not in the desired format"`

Reply:
They aren't the same for transcoding at least, value will be the enum and text is the description the user sees. For codec selection the value and text are indeed the same.

One more thing and this was already wrong before but: the `for` attribute of label should link to the `id` of select, now those two values are different. Basically to fix this, also set `<label for="{name}-select"`

Reply:
I was only seeing the `targetVideoCodec` section, my bad!

Reply:
> Basically to fix this, also set `<label for="{name}-select"`

Done!